src/collection-view.js

```diff
@@ -30,6 +26,11 @@ Marionette.CollectionView = Marionette.View.extend({
     Marionette.View.apply(this, arguments);
 
     this.on('show', this._onShowCalled);
+
+    if (_.isUndefined(this.sort)) {
+        _.defaults(this.options, { sort: true });
+    }
```

Comment:
Yeah I agree.  This part of this PR was a bit of a rabbit trail, so I was attempting to keep it similar, but I could help making at least more readable.
I'm all for putting sort on the prototype. That would replace this entire block.  
As for #3, I think that'd be useful assuming it sticks around, though we wouldn't need it here.